lib/token.js

```diff
@@ -379,26 +379,44 @@ function getIdToken(sdk, oauthOptions, options) {
   function handleOAuthResponse(res) {
     if (res['error'] || res['error_description']) {
       throw new OAuthError(res['error'], res['error_description']);
+    }
 
-    } else if (res['id_token']) {
-      if (res.state !== oauthParams.state) {
-        throw new AuthSdkError('OAuth flow response state doesn\'t match request state');
-      }
+    if (res.state !== oauthParams.state) {
+      throw new AuthSdkError('OAuth flow response state doesn\'t match request state');
+    }
+    
+    var constructedToken = {
+      scopes: oauthQueryHash.scope.split(' ')
+    };
 
+    if (res['id_token']) {
       var jwt = sdk.idToken.decode(res['id_token']);
       if (jwt.payload.nonce !== oauthParams.nonce) {
         throw new AuthSdkError('OAuth flow response nonce doesn\'t match request nonce');
       }
 
       validateClaims(jwt.payload, sdk.options.url, oauthParams.clientId);
-      return {
+
+      util.extend(constructedToken, {
         idToken: res['id_token'],
-        claims: jwt.payload
-      };
+        claims: jwt.payload,
+        expiresAt: jwt.payload.exp
+      });
+    }
+    
+    if (res['access_token']) {
+      util.extend(constructedToken, {
+        accessToken: res['access_token'],
+        expiresAt: Number(res['expires_in']) + Math.floor(Date.now()/1000),
```

Comment:
We will always have a payload.exp for the idToken, but we can't parse the access_token. We have to get expires_in from the oauth response (in the postMessage or hash fragment) and calculate expiresAt.

This will overwrite any expiresAt copied from the idToken. It's possible to remove expiresAt entirely for the idToken, because it does duplicate the information and it'd avoid confusion.